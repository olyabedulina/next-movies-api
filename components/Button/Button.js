import React from 'react'
import PropTypes from 'prop-types'

import CM from './styles.module.css'

const Button = ({
    type = 'button',
    className = '',
    kind = 'main',
    children,
    onClick = Function.prototype
}) => {
    function handleClick() {
        onClick();
    }

    return <button
        className={`${className} ${CM.button} ${CM["kind-" + kind] || ''}`}
        type={type}
        onClick={handleClick}>
            {children}
        </button>
}

Button.propTypes = {
    type: PropTypes.string,
    className: PropTypes.string,
    kind: PropTypes.string,
    children: PropTypes.node,
    onClick: PropTypes.func
};

export default Button
