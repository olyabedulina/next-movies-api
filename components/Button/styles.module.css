.button {
  display: inline-block;
  outline: 0;
  overflow: hidden;
  margin: 0;
  padding: 0;
  border: none;
  cursor: pointer;
  text-transform: uppercase;
}

.kind-main {
  box-sizing: border-box;
  border-radius: 10px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 16px 56px;
  font-size: 18px;
  line-height: 19px;
  background: #f65261;
  color: #fff;
}

.kind-alt {
  padding: 16px 56px;
  border-radius: 6px;
  font-size: 18px;
  background-color: rgba(35, 35, 35, 0.3);
  border: 1px solid #f65261;
  color: #f65261;
}

.kind-aux {
  padding: 16px 18px;
  border-radius: 6px;
  font-size: 18px;
  background-color: rgba(255, 255, 255, 0.3);
  color: #f65261;
}