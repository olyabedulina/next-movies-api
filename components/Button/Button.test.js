import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('Components :: Button snapshots:', () => {
    it('Test :: "+ Add Movie" Button component', () => {
        const actual = renderer
            .create(<Button kind="aux" className="buttonAddMovie">+ Add Movie</Button>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: "Search" Button component', () => {
        const actual = renderer
            .create(<Button type="submit" className="formButton">Search</Button>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: "Reset" Button component', () => {
        const actual = renderer
            .create(<Button kind='alt' className="modalFooterButton">Reset</Button>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: "Submit" Button component', () => {
        const actual = renderer
            .create(<Button type="submit" kind='main' className="modalFooterButton">Submit</Button>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: "Go Back To Home" Button component', () => {
        const actual = renderer
            .create(<Button kind='alt' className="pageNotFoundButton">Go Back To Home</Button>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });
});
