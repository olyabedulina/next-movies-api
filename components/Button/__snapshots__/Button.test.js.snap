// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components :: Button snapshots: Test :: "+ Add Movie" Button component 1`] = `
<button
  className="buttonAddMovie button kind-aux"
  onClick={[Function]}
  type="button"
>
  + Add Movie
</button>
`;

exports[`Components :: Button snapshots: Test :: "Go Back To Home" Button component 1`] = `
<button
  className="pageNotFoundButton button kind-alt"
  onClick={[Function]}
  type="button"
>
  Go Back To Home
</button>
`;

exports[`Components :: Button snapshots: Test :: "Reset" Button component 1`] = `
<button
  className="modalFooterButton button kind-alt"
  onClick={[Function]}
  type="button"
>
  Reset
</button>
`;

exports[`Components :: Button snapshots: Test :: "Search" Button component 1`] = `
<button
  className="formButton button kind-main"
  onClick={[Function]}
  type="submit"
>
  Search
</button>
`;

exports[`Components :: Button snapshots: Test :: "Submit" Button component 1`] = `
<button
  className="modalFooterButton button kind-main"
  onClick={[Function]}
  type="submit"
>
  Submit
</button>
`;
